src/components/chat/chat_messages.jsx

```diff
@@ -27,21 +25,23 @@ const ChatMessages = React.createClass({
     }
   },
 
-  componentWillReceiveProps() {
+  componentWillReceiveProps(p) {
     let previousChannel = this.state.channel;
+
     this.setState({
       channel: this.getParams().roomSlug
     }, () => {
-      if (previousChannel !== this.state.channel) {
-        ChatActions.destroy();
-        ChatActions.init(
-          `${AppStore.getUrl()}/rooms/${this.state.channel}/messages`,
-          CurrentUserStore.getToken()
-        );
-      }
+      this.updateMessages();
+      this.handleChannelChange()
     });
   },
 
+  handleChannelChange(){
```

Comment:
So in that case, what should be the behavior when a user goes directly to
`/rooms/rippratchett` without having joined that room? Should we show a big
buton in the center of the screen saying "Join this room"?

On Mon, Mar 16, 2015 at 5:46 PM Charles Pletcher <notifications@github.com>
wrote:

> In src/components/chat/chat_messages.jsx
> <https://github.com/asm-products/landline-web/pull/4#discussion_r26503185>
> :
>
> >      });
> >    },
> >
> > +  handleChannelChange(){
>
> I think we can take the Slack route and say that a user has to subscribe
> to a room in order to enter it -- so they'll be subscribed to every room
> that they're in.
>
> A slightly more complicated trigger might involve checking whether the
> length of the messages List is 0 and whether we've received a response
> for the current room from the server.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asm-products/landline-web/pull/4/files#r26503185>.
>